src/main/java/com/lineate/testyourlexicon/services/implementations/UserServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.lineate.testyourlexicon.services.implementations;
+
+import com.lineate.testyourlexicon.dto.UserRegistrationDto;
+import com.lineate.testyourlexicon.entities.User;
+import com.lineate.testyourlexicon.repositories.UserRepository;
+import com.lineate.testyourlexicon.services.UserRegistrationResponse;
+import com.lineate.testyourlexicon.services.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+
+  private final UserRepository userRepository;
+  private final PasswordEncoder passwordEncoder;
+
+  public UserRegistrationResponse createUser(UserRegistrationDto userRegistrationDto) {
+    if (userRepository.findUserByEmail(userRegistrationDto.getEmail()).isPresent()) {
```

Comment:
Let's set up the unique constraint on the email column in the database. In this case, an exception will be thrown in when we try to add a user with an already existing email. We can then catch this exception in the service and throw a validation exception, which can be handled by @ExceptionHandler annotation. This way we don't need if statements and don't need to have UserRegistrationResponse